logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -148,7 +148,10 @@ def initialize(queue, size, wait)
         @shutdown_signal_received = false
         @flush_signal_received = false
         @originals = Hash.new
-        @cancelled = Hash.new
+
+        # disabled for https://github.com/elastic/logstash/issues/6055
+        # @cancelled = Hash.new
```

Comment:
I would leave it for now since this is a quick fix and we may come back to it and "redo" this for some reason